lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -906,7 +906,7 @@ protected function handle_featured_media( $featured_media, $post_id ) {
 			if ( $result ) {
 				return true;
 			} else {
-				return new WP_Error( 'rest_invalid_featured_media', __( 'Invalid featured media id.' ), array( 'status' => 400 ) );
+				return new WP_Error( 'rest_invalid_featured_media', __( 'Invalid featured media id.' ), array( 'status' => 404 ) );
```

Comment:
This should remain a `400` error because the provided featured_image id doesn't exist and the request needs to be modified.